### DISCUSSION TITLE
为什么await能防止InfiniteScroll 无限调用呢？

### DISCUSSION THREAD

**Discussion by githubqh** [ant-design/ant-design-mobile#5455] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/105548468/177902382-f3cfd24d-7c6e-4ab3-94de-05c732f32414.png)
![image](https://user-images.githubusercontent.com/105548468/177902444-a84fb4da-1a6b-4bbf-9180-08e5ccab7dc6.png)

另外想问一下这两个函数不等效吗？
![image](https://user-images.githubusercontent.com/105548468/177901399-a47e055e-d6b9-40df-9023-d8e185462f0e.png)

```js
async function log(time) {
    setTimeout(function () {
        console.log(time)
        return 1
    }, time)
}
function log(time) {
    return new Promise((resolve, reject) => {
        setTimeout(function () {
            console.log(time)
            resolve(1)
        }, time)
    })
}

async function fun() {
    console.log(await log(2000));
    log(1000)
    console.log(1)
}
fun()
```
另外，为什么下面两个写法是正确的？
![image](https://user-images.githubusercontent.com/105548468/177902109-6a7a07f2-3066-4e2f-ab9e-d7cf2069d7dc.png)

**Comment by IronKinoko** [comment, marked as answer]
你提供的两个 log 函数不是等效的，可以阅读 [async 文档](https://developer.mozilla.org/zh-CN/docs/Web/JavaScript/Reference/Statements/async_function)
